### DISCUSSION TITLE
removing HTML from errors at top of CRUD page

### DISCUSSION THREAD

**Discussion by prescriptionlifeline** [Laravel-Backpack/community-forum#1018] [Q&A (Help)] [answered]
In my CRUD controller, in `setupCreateOperation()`, I have this:

```
CRUD::setValidation(NewUserRequest::class);
```
Here are the validation rules for the email field in `NewUserRequest`:

```php
'email' => ['required', 'string', 'lowercase', 'email', 'max:255', new Email],
```
Here's the Email class:

```php
<?php

namespace App\Rules;

use Closure;
use Illuminate\Contracts\Validation\ValidationRule;
use App\Models\User;

class Email implements ValidationRule
{
    /**
     * Run the validation rule.
     *
     * @param  \Closure(string): \Illuminate\Translation\PotentiallyTranslatedString  $fail
     */
    public function validate(string $attribute, mixed $value, Closure $fail): void
    {
        $user = User::where('email', $value)->withTrashed()->first();
        if ($user) {
            if (!$user->deleted_at) {
                $fail("A user with this email address <a target='_blank' href='/admin/user/{$user->id}/edit'>already exists</a>.");
            } else {
                $user->restore();
                $fail("A user with this email address had been soft-deleted. They have <a target='_blank' href='/admin/user/{$user->id}/edit'>now been restored</a>.");
            }
        }
    }
}
```
So, basically, if you're trying to create a new user with the same email as the existing user you'll be linked to the old user. This works well save for one thing:

![Screenshot 2024-06-09 115323](https://github.com/Laravel-Backpack/community-forum/assets/77934157/6e6fa1f3-98d6-46d7-b4d0-bb51efc253b2)
So the error that appears under the problem field shows the text with the hyperlink intact but the error message that appears at the top appears to have been run through `htmlspecialchars()` or some such. I don't want it to show any HTML at all. If it shows a hyperlink that's cool, if it doesn't, that's cool, too, but I don't like the way it's currently displaying. In vanilla PHP I could do `htmlspecialchars_decode` or `strip_tags` but I'm operating in the context of a validation rule being called by a FormRequest called by a CRUD controller.

Any ideas?

If there's a better way to achieving the same effect that'd be good to know as well!

**Comment by jcastroa87** [comment, marked as answer]
Hello @prescriptionlifeline 

You have this options:

1. Disable "groupedErrors", in /config/backpack/operations/create.php
```
'groupedErrors' => false,
```

2. You can change how the create.blade.php file looks and works, by just placing a file with the same name in your resources/views/vendor/backpack/crud/create.blade.php. To quickly do that, run:
```
php artisan backpack:publish crud/create
```
And then customize as you need
```
@include('crud::inc.grouped_errors')
```

Cheers.

**Reply by prescriptionlifeline** [reply]
Building off of this I did `php artisan backpack:publish crud/inc/grouped_errors` and then replaced `{{ $message }}` with `{!! $message !!}`.

**Reply by prescriptionlifeline** [reply]
One thing I'm curious about...  so `php artisan backpack:publish crud/inc/grouped_error` copies files from the `vendor/backpack/crud/src/resources/views` directory but what if I wanted to copy files from `vendor/backpack/theme-tabler/resources/views` instead? Do I just need to copy those files manually or is there a `php artisan` command that can be used for that one as well? https://stackoverflow.com/a/78134209/569976 describes an scenario where one might want to do that.

Thanks!

**Reply by jcastroa87** [reply]
Hello @prescriptionlifeline 

There is not php artisan command instead you can achieve with this:

```
# create the custom directory if it's not already there
mkdir -p resources/views/vendor/backpack/theme-tabler

# copy the blade file inside the folder we created above
cp -i vendor/backpack/crud/src/resources/views/ui/dashboard.blade.php resources/views/vendor/backpack/theme-tabler/dashboard.blade.php
```

Cheers.

**Comment by jcastroa87** [comment]
I will close the issue, but please, feel free to re-open or create a new one if needed.
